tests/auxiliary/minisimd.rs

```diff
@@ -147,14 +146,100 @@ impl<T, const N: usize> PackedSimd<T, N> {
     pub const fn from_array(a: [T; N]) -> Self {
         PackedSimd(a)
     }
-    pub fn as_array(&self) -> &[T; N] {
+    pub const fn as_array(&self) -> &[T; N] {
         let p: *const Self = self;
         unsafe { &*p.cast::<[T; N]>() }
     }
-    pub fn into_array(self) -> [T; N]
+    pub const fn into_array(self) -> [T; N]
     where
         T: Copy,
     {
         *self.as_array()
     }
+    pub const fn splat(a: T) -> Self
+    where
+        T: Copy,
+    {
+        Self([a; N])
+    }
 }
+
+// As `const_trait_impl` is a language feature with specialized syntax, we have to use them in a way
+// such that it doesn't get parsed as Rust code unless `cfg(minisimd_const)` is on. The easiest way
+// for that is a macro
+
+#[cfg(minisimd_const)]
+macro_rules! impl_traits {
+    () => {
+        impl<T: [const] PartialEq, const N: usize> const PartialEq for Simd<T, N> {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+
+        impl<T, const N: usize> const core::ops::Index<usize> for Simd<T, N> {
+            type Output = T;
+            fn index(&self, i: usize) -> &T {
+                &self.as_array()[i]
+            }
+        }
+
+        impl<T: [const] PartialEq, const N: usize> const PartialEq for PackedSimd<T, N> {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+    };
+}
+
+#[cfg(not(minisimd_const))]
+macro_rules! impl_traits {
+    () => {
+        impl<T: PartialEq, const N: usize> PartialEq for Simd<T, N> {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+
+        impl<T, const N: usize> core::ops::Index<usize> for Simd<T, N> {
+            type Output = T;
+            fn index(&self, i: usize) -> &T {
+                &self.as_array()[i]
+            }
+        }
+
+        impl<T: PartialEq, const N: usize> PartialEq for PackedSimd<T, N> {
+            fn eq(&self, other: &Self) -> bool {
+                self.as_array() == other.as_array()
+            }
+        }
+    };
+}
+
+impl_traits!();
```

Comment:
Nit: Since we're using a macro here anyhow, you can deduplicate with:
```suggestion
macro_rules! impl_traits {
    ($($const_:ident)?) => {
        impl<T: $([$const_])? PartialEq, const N: usize> const PartialEq for Simd<T, N> {
            fn eq(&self, other: &Self) -> bool {
                self.as_array() == other.as_array()
            }
        }

        impl<T, const N: usize> $($const_)? core::ops::Index<usize> for Simd<T, N> {
            type Output = T;
            fn index(&self, i: usize) -> &T {
                &self.as_array()[i]
            }
        }

        impl<T: $([$const_])? PartialEq, const N: usize> const PartialEq for PackedSimd<T, N> {
            fn eq(&self, other: &Self) -> bool {
                self.as_array() == other.as_array()
            }
        }
    };
}

#[cfg(minisimd_const)]
impl_traits!(const);

#[cfg(not(minisimd_const))]
impl_traits!();
```